src/test/java/com/github/martapietka/ppm2pbmSpring/ImageServiceTest.java

```diff
@@ -0,0 +1,54 @@
+package com.github.martapietka.ppm2pbmSpring;
+
+
+import org.assertj.core.api.Assertions;
+import org.junit.jupiter.api.Test;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+
+import java.io.InputStream;
+
+@SpringBootTest
+class ImageServiceTest {
+
+    @Autowired
+    private ImageService imageService;
+
+    @Test
+    public void testGetConvertedInputStream() throws Exception {
+
+        String imageName = "3x3.ppm";
+
+        String expectedOutputString = """
+                P1
+                3 3
+                1 0 1
+                0 1 0
+                0 0 1
+                """;
+
+        try (InputStream inputStream = getClass().getResourceAsStream("/" + imageName)) {
+
+            InputStream convertedInputStream = imageService.getConvertedInputStream(inputStream);
+            byte[] convertedBytes = convertedInputStream.readAllBytes();
+
+            Assertions.assertThat(convertedBytes).asString().isEqualTo(expectedOutputString);
+        }
+    }
+
+//    @Test
+//    public void testGetImage() throws IOException {
```

Comment:
Tutaj jest prawdziwy problem - `ImageIO.write` jest metodą statyczną, więc nie da się jej łatwo zamockować (jest to możliwe przy użyciu dodatkowych bibliotek typu PowerMock, ale raczej odradza się ich stosowanie). Co można zrobić?

Najlepiej rozdzielić odpowiedzialność klasy i stworzyć jakąś osobną, np. `PnmToPngConverter` która ma tę metodę i dla niej napisać osobny test albo... odpuścić :) niektóre testy można uzupełnić później albo w ostateczności odpuścić (chociaż akurat to nie jest zalecane z oczywistych względów) jeżeli mają niewiele logiki, a koncentrują się na wywołaniu czegoś, co trudno zamockować (zewnętrzne statyczne metody, zewnętrzne API, bazy danych, itd.).

Kluczowe jest przede wszystkim przetestowanie Twojej logiki aplikacyjnej.